### DISCUSSION TITLE
@Acknowledgment(Acknowledgment.Strategy.MANUAL) doesnt seem to work

### DISCUSSION THREAD

**Discussion by samarism** [quarkusio/quarkus#43927] [Community]
I am using this @Acknowledgment(Acknowledgment.Strategy.MANUAL) and sending the ack manually after processin the record. BUT kafka keeps on sending messages without waiting for my ack. Why is it so? Please help. Thanks.

**Comment by samarism** [comment]
Assumption: quarkus should not call poll() method till i am done with processing the previous batch and have sent the acknoledgements for all the messages in the previous batch. 

But seems that quarkus keeps calling poll() method, cause i keep on receiving messages.

**Reply by cescoffier** [reply]
It will continue calling `poll` until it runs _out of requests_. 

Check https://smallrye.io/smallrye-reactive-messaging/latest/kafka/receiving-kafka-records/#acknowledgement and https://smallrye.io/smallrye-reactive-messaging/latest/kafka/writing-kafka-records/#back-pressure-and-inflight-records
